### PR TITLE
feat: migrate to ESM

### PR DESCRIPTION
**Pull request by wolfy1339**
BREAKING CHANGE: Migrate to ESM builds

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,9 @@
     }
   },
   "compilerOptions": {
+    "lib": ["ESNext"],
```
Comment (wolfy1339) [inline tsconfig.json:10]: These can be moved to `@octokit/tsconfig` later on

File: package.json [hunk 2]
```diff
@@ -43,14 +44,13 @@
     "@octokit/request-error": "^5.0.0",
     "@octokit/webhooks-methods": "^4.0.0",
     "@wolfy1339/openapi-webhooks-types": "5.1.3",
-    "aggregate-error": "^3.1.0"
+    "aggregate-error": "^5.0.0"
```
Comment (wolfy1339) [inline package.json:47]: While `AggregateError` was introduced natively in NodeJS v15, it isn't generic. So the types cannot be extended to use our custom error

File: src/types.ts [hunk 3]
```diff
@@ -75,20 +76,3 @@ export interface WebhookEventHandlerError<TTransformed = unknown>
     ? EmitterWebhookEvent
     : EmitterWebhookEvent & TTransformed;
 }
-
```
Comment (wolfy1339) [inline src/types.ts:78]: The types were fixed in version 3.0.1 and these are out of date versus v5 of `aggregate-error`

File: test/integration/event-handler-test.ts [hunk 4]
```diff
@@ -91,12 +91,12 @@ describe("when a handler throws an error", () => {
     } catch (error: any) {
       expect(error.message).toMatch(/oops/);
 
-      const errors = Array.from(error);
+      const errors = Array.from(error.errors);
```
Comment (wolfy1339) [inline test/integration/event-handler-test.ts:94]: The `AggregateError` class doesn't extend `Iterable<T>` anymore, you have to use the `.errors` property

**Review by nickfloyd — APPROVED** [review]
All of this looks good @wolfy1339 - thank you for pushing this forward ❤️.  Just a clarifying question  - are you planning a follow along set of changes in octokit.js and app.js for imports and such since those are still cjs modules?

**Comment by wolfy1339** [comment]
Yes, I will be doing the other modules as well.

**Pull request merged**